journal/journal/views.py

```diff
@@ -0,0 +1,85 @@
+import transaction
+from pyramid.response import Response
+from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPFound
+from wtforms import Form, StringField, TextAreaField, validators
+from jinja2 import Markup
+import markdown
+
+from sqlalchemy.exc import DBAPIError
+
+from .models import (
+    DBSession,
+    Entry,
+    )
+
+
+class NewEntry(Form):
+    title = StringField('title', [validators.Length(min=1, max=128)])
+    text = TextAreaField('text')
+
+
+@view_config(route_name='home', renderer='templates/home.jinja2')
+def my_view(request):
+    try:
+        all_entries = DBSession.query(Entry).order_by(Entry.id.desc()).all()
+        return {'entries': all_entries}
+    except DBAPIError:
+        return Response("home broke", content_type='text/plain', status_int=500)
+
+
+@view_config(route_name='entry', renderer='templates/entry.jinja2')
+def entry_detail(request):
+    try:
+        entry_id = request.matchdict['entry_id']
+        entry = DBSession.query(Entry).get(entry_id)
+        entry.text = render_markdown(entry.text)
```

Comment:
You are mutating the `text` attribute of your entry here and turning it from raw markdown into rendered HTML.  You don't want to do that.  If the request returns properly, then the change will be permanent.